cmd/doctor.go

```diff
@@ -0,0 +1,92 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/spf13/cobra"
+	vfs "github.com/twpayne/go-vfs"
+)
+
+// doctorCmd represents the doctor command
+var doctorCommand = &cobra.Command{
+	Use:   "doctor",
+	Short: "Check your system for potential problems",
+	Long: `Check your system for potential problems. Such as:
+* Does the configuration file exist
+* Does the destination directory exist
+* Does the source directory exist
+* Is Hashicorp Vault installed?
+* Is Bitwarden installed?
+* Is LastPass installed?
+`,
+	RunE: makeRunE(config.runDoctorCommand),
+}
+
+func init() {
+	rootCommand.AddCommand(doctorCommand)
+
+}
+
+func (c *Config) runDoctorCommand(fs vfs.FS, args []string) error {
+	fmt.Printf("Arch: %s\n", runtime.GOARCH)
+	fmt.Printf("OS: %s\n", runtime.GOOS)
+
+	info, err := os.Stat(configFile)
+	if os.IsNotExist(err) {
+		fmt.Printf("Configuration File: Does Not Exist (%s)\n", configFile)
+	} else {
+		fmt.Printf("Configuration File: Exists (%s %s)\n", info.Mode().Perm().String(), configFile)
+	}
+
+	info, err = os.Stat(config.DestDir)
+	if os.IsNotExist(err) {
+		fmt.Printf("Destination Directory: Does Not Exist (%s)\n", config.DestDir)
+	} else {
+		fmt.Printf("Destination Directory: Exists (%s %s)\n", info.Mode().Perm().String(), config.DestDir)
+	}
+
+	info, err = os.Stat(config.SourceDir)
+	if os.IsNotExist(err) {
+		fmt.Printf("Source Directory: Does Not Exist (%s)\n", config.SourceDir)
+	} else {
+		fmt.Printf("Source Directory: Exists (%s %s)\n", info.Mode().Perm().String(), config.SourceDir)
+	}
+
+	path, err := exec.LookPath(config.Vault.Vault)
+	if err == nil {
+		info, err = os.Stat(path)
+		if !os.IsNotExist(err) {
+			fmt.Printf("Hashicorp Vault: Installed (%s %s)\n", info.Mode().Perm().String(), path)
+		} else {
+			fmt.Printf("Hashicorp Vault: Error (%s)\n", err.Error())
+		}
+	} else {
+		fmt.Println("Hashicorp Vault: Not Found")
+	}
+	path, err = exec.LookPath(config.Bitwarden.BW)
+	if err == nil {
+		info, err = os.Stat(path)
+		if !os.IsNotExist(err) {
+			fmt.Printf("Bitwarden: Installed (%s %s)\n", info.Mode().Perm().String(), path)
+		} else {
+			fmt.Printf("Bitwarden: Error (%s)\n", err.Error())
+		}
+	} else {
+		fmt.Println("Bitwarden: Not Found")
+	}
+	path, err = exec.LookPath(config.LastPass.Lpass)
+	if err == nil {
+		info, err = os.Stat(path)
+		if !os.IsNotExist(err) {
+			fmt.Printf("LastPass: Installed (%s %s)\n", info.Mode().Perm().String(), path)
+		} else {
+			fmt.Printf("LastPass: Error (%s)\n", err.Error())
+		}
+	} else {
+		fmt.Println("LastPass: Not Found")
+	}
```

Comment:
Please refactor these three tests into a `for`-loop over a slice, something like:

```go
    for _, binary := range []struct{
        description string
        name        string
    }{
        {description: "Bitwarden", name: config.Bitwarden.BW},
        {description: "LastPass", name: config.LastPass.Lpass},
        {description: "Vault", name: config.Vault.Vault},
    } {
        if path, err := exec.LookPath(binary.name); err == nil {
            // ...
        } else {
            // ...
        }
    }
```

Reply:
Updated